server/src/main/java/org/opensearch/tracing/TracerModule.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.plugins.TracerPlugin;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * A module for loading classes for tracer
+ *
+ * @opensearch.internal
+ */
+public class TracerModule {
+
+    private static final String TRACER_TYPE_DEFAULT_KEY = "tracer.type.default";
+    private static final String TRACER_TYPE_KEY = "tracer.type";
+
+    public static final Setting<String> TRACER_DEFAULT_TYPE_SETTING = Setting.simpleString(
+        TRACER_TYPE_DEFAULT_KEY,
+        Setting.Property.NodeScope
+    );
+    public static final Setting<String> TRACER_TYPE_SETTING = Setting.simpleString(TRACER_TYPE_KEY, Setting.Property.NodeScope);
+
+    private final Settings settings;
+    private final Map<String, Supplier<Tracer>> tracerFactories = new HashMap<>();
+    private final Map<String, TracerHeaderInjector> tracerHeaderInjectors = new HashMap<>();
+
+    public TracerModule(Settings settings, List<TracerPlugin> tracerPlugins, ThreadPool threadPool, TracerSettings tracerSettings) {
+        this.settings = settings;
+
+        for (TracerPlugin tracerPlugin : tracerPlugins) {
+            Map<String, Supplier<Tracer>> tracerFactory = tracerPlugin.getTracers(threadPool, tracerSettings);
```

Comment:
Are there multiple tracers supported at a time? shouldn't we ensure single tracer to be enabled?

Reply:
We do ensure that we support only one tracer at any time. We set the default tracer span in Otel plugin

```
@Override
    public Settings additionalSettings() {
        if (FeatureFlags.isEnabled(TRACER)) {
            return Settings.builder()
                // set Otel tracer as default tracer
                .put(TracerModule.TRACER_DEFAULT_TYPE_SETTING.getKey(), OTEL_TRACER_NAME)
                .build();
        }
        return Settings.EMPTY;
    }
```